src/persist-client/src/cli/admin.rs

```diff
@@ -125,10 +134,115 @@ pub async fn run(command: AdminArgs) -> Result<(), anyhow::Error> {
             .await?;
             info_log_non_zero_metrics(&metrics_registry.gather());
         }
+        Command::RestoreBlob(args) => {
+            let StoreArgs {
+                consensus_uri,
+                blob_uri,
+            } = args;
+            let commit = command.commit;
+            let cfg = PersistConfig::new(&BUILD_INFO, SYSTEM_TIME.clone());
+            let metrics_registry = MetricsRegistry::new();
+            let metrics = Arc::new(Metrics::new(&cfg, &metrics_registry));
+            let consensus =
+                make_consensus(&cfg, &consensus_uri, commit, Arc::clone(&metrics)).await?;
+            let blob = make_blob(&cfg, &blob_uri, commit, Arc::clone(&metrics)).await?;
+            let versions = StateVersions::new(
+                cfg.clone(),
+                Arc::clone(&consensus),
+                Arc::clone(&blob),
+                metrics,
+            );
+
+            let mut shards = consensus.list_keys();
+            let mut not_restored = vec![];
+            while let Some(shard) = shards.next().await {
```

Comment:
is it crazy to run restore_blob concurrently? feels like we might want some amount of concurrency in the tool for big envs and this might be an easy place to to do it. otoh, I'm 100% okay leaving this as a TODO

Reply:
👍 I think what I'm worried about here is that we end up in an incident and find out that the restore tool is gonna take many hours because we don't have any sort of concurrency. If we had some data that running this on a huge env still finished in O(minutes), then I definitely don't care. Fine with making it a cli arg, but I'd lean toward making the default of that arg to be some amount of concurrency, unless we have evidence that things tend to be fast enough even without it